src/index.js

```diff
@@ -148,14 +125,26 @@ function getInterface(proto) {
  */
 function getPropertyNames(proto) {
     const results = []
-
     while (proto !== Object.prototype) {
         Object.getOwnPropertyNames(proto).reduce((arr,k) => arr.indexOf(k) === -1 ? arr.push(k) && arr : arr, results)
         proto = Object.getPrototypeOf(proto).constructor.prototype
     }
-
     return results
 }
 
+function isMixin(x) {
+    return (typeof x == 'function') && !!x.mixin
+}
+
+function isClass(x) {
+    return (typeof x == 'function') && (s => /^class\s/.test(s)
+        || /^.*classCallCheck\(/.test(s.replace(/^[^{]*{\s*/,'').replace(/\s*}[^}]*$/,'')))(x.toString())
```

Comment:
Ok but that is a different issue. This was about removing the type strings (`'class'`, `'factory'` etc) support from `is()` and `like()`.